src/devices/tpms_aston_martin.c

```diff
@@ -0,0 +1,151 @@
+
+Preamble/Syncword  .... : 0x32b4
+
+    6 bytes message
+    Byte Position   0  1  2  3  4  5
+    Sample         28 3f ff ff 00 fa
+                   VV |I II II || CC
+                      |        ||
+                 +----+-+  +----------+
+                 | 1234 |  | 12345678 |
+                 | MMII |  | FFFFFFXX |
+                 +------+  +----------+
+
+- VV: {8} Pressure value, offset 40, scale 2.5 if Message Type = 0x00
+          Temperature Value, offset 40,        if Message Type = 0x01
+- MM: {2} Message Type, 0x00 or 0x01
+- II:{22} Sensor ID,
+- F : {8} Flags, F1 = quick inflate detected,
+                 F2 to F6 are unknown,
+                 X78:{2} Looks like XOR/checksum/Parity from previous bits, not decoded.
+- C : {7} CRC-7, poly 0x45, init 0x6f, final XOR 0x00
+
+Bitbench:
+
+    TEMP/PRESSURE 8d MESSAGE_TYPE 2b ID 22d FAST_INCREASE 1b FLAGS ? 5b PARITY_XOR ? 2b CRC_SEVEN 7b 1x
+
+*/
+
+static int tpms_aston_martin_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    bitbuffer_t decoded = { 0 };
+    uint8_t *b;
+    uint8_t const preamble_pattern[] = {0x32, 0xb4};
+    uint8_t len_msg = 6;
+    float pressure_kPa = 0;
+    int temperature_C = 0;
+
+    if (bitbuffer->num_rows != 1) {
+        return DECODE_ABORT_EARLY;
+    }
+
+    int pos = bitbuffer_search(bitbuffer, 0, 0, preamble_pattern, sizeof(preamble_pattern) * 8);
+    if (pos >= bitbuffer->bits_per_row[0]) {
+        decoder_logf(decoder, 1, __func__, "Preamble not found");
+        return DECODE_ABORT_EARLY;
+    }
+
+    decoder_log_bitrow(decoder, 1, __func__, bitbuffer->bb[0], bitbuffer->bits_per_row[0], "MSG");
+    bitbuffer_differential_manchester_decode(bitbuffer, 0, pos + sizeof(preamble_pattern) * 8, &decoded, len_msg * 8);
+    decoder_log_bitrow(decoder, 1, __func__, decoded.bb[0], decoded.bits_per_row[0], "DMC");
+
+    // check msg length
+    if (decoded.bits_per_row[0] < (len_msg * 8) - 1 ) { // always missing last bit
+        decoder_logf(decoder, 1, __func__, "Too short");
+        return DECODE_ABORT_LENGTH;
+    }
+
+    b = decoded.bb[0];
+
+    // verify checksum
+    if (crc7(b, len_msg, 0x45, 0x6f)) {
+        decoder_logf(decoder, 1, __func__, "crc error");
+        return DECODE_FAIL_MIC; // crc mismatch
+    }
+
+    int id       = ((b[1] & 0x3f) << 16) | (b [2] << 8) | b[3];
+    int msg_type = (b[1] & 0xc0) >> 6;
+    int value    = b[0] - 40;
+
+    if (msg_type == 0) { // pressure
+        pressure_kPa = value * 2.5;
+    }
+    else if (msg_type == 1) { // temperature
+        temperature_C = value;
+    }
+    else {
+        decoder_logf(decoder, 1, __func__, "Unknown message type %x", msg_type);
+        return DECODE_ABORT_EARLY;
+    }
+
+    int inflate = (b[4] & 0x80) >> 7;
+    int flags   = b[4] & 0x7f;
+
+    /* clang-format off */
+    data_t *data = data_make(
+            "model",         "",              DATA_STRING, "Aston-Martin-Smartire",
```

Comment:
Only a single dash, and should be shorter. Since SmarTire is the manufacturer maybe `SmarTire-AM`?
And decoder file could be tpms_smartire.c perhaps?

Reply:
you're right, renamed everywhere.